modules/infra/src/main/scala/scaladex/infra/SqlDatabase.scala

```diff
@@ -48,11 +49,11 @@ class SqlDatabase(datasource: HikariDataSource, xa: doobie.Transactor[IO]) exten
   ): Future[Boolean] = {
     val unknownStatus = GithubStatus.Unknown(time)
     for {
-      isNewProject <- insertProjectRef(artifact.projectRef, unknownStatus)
+      maybeIsNewProject <- artifact.projectRef.traverse(ref => insertProjectRef(ref, unknownStatus))
       _ <- run(ArtifactTable.insertIfNotExist(artifact))
-      _ <- run(ReleaseTable.insertIfNotExists.run(artifact.release))
+      _ <- artifact.release.traverse(release => run(ReleaseTable.insertIfNotExists.run(release)))
       _ <- insertDependencies(dependencies)
-    } yield isNewProject
+    } yield maybeIsNewProject.fold(true)(identity)
```

Comment:
@adpi2 I hope you don't mind me tagging you for some early feedback.

This is the part that I'm most iffy about. What do we consider a new artifact now, with the ability to store artifacts that don't _immediately_ have an associated project?

The result of this `maybeIsNewProject` is used in [a conditional](https://github.com/scalacenter/scaladex/pull/1013/files#diff-8c8a9fdd0e802d7eeb93efa0e7420ad297fe446a0b2eb77639ae84c1ff43179fR89) later to determine whether to update the GitHub project information.

Reply:
It looks okay to return false if an artifact has no associated project.